pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -976,21 +939,22 @@ func setBridgeIPAddr(
 			status.BridgeName)
 		return errors.New(errStr)
 	}
-	if status.Gateway != nil {
-		ipAddr = status.Gateway.String()
-		status.IPAssignments[bridgeMac.String()] = status.Gateway
-	}
 	log.Functionf("BridgeMac: %s, ipAddr: %s\n",
 		bridgeMac.String(), ipAddr)
-	status.BridgeIPAddr = ipAddr
-	publishNetworkInstanceStatus(ctx, status)
-	log.Functionf("Published NetworkStatus. BridgeIpAddr: %s\n",
-		status.BridgeIPAddr)
 
-	if status.BridgeIPAddr == "" {
-		log.Functionf("Does not yet have a bridge IP address for %s\n",
-			status.Key())
-		return nil
+	// outside of subnet, flag it
+	if !status.Subnet.Contains(status.Gateway) {
+		errStr := fmt.Sprintf("Bridge IP is not in Subnet")
```

Comment:
In this case it is the Gateway which is not in subnet. Would be helpful to add the Gateway IP to the error message.

Reply:
done